bam/src/main/java/com/hartwig/bam/after/BamIndexPipeline.java

```diff
@@ -31,24 +33,31 @@ private BamIndexPipeline(final FileSystem fileSystem, final String bamFolder, fi
         this.sortAndIndexer = sortAndIndexer;
     }
 
-    public void execute(Sample sample) throws IOException, InterruptedException {
-        String bamFileLocation = String.format("/%s/%s.bam", sourceBamDirectory, sample.name()).substring(1);
-        String unsortedBam = Bams.name(sample, localBamDirectory, "unsorted");
+    public void execute(Sample sample, StatusReporter statusReporter) {
```

Comment:
Why pass the reporter as an argument here when we use it as a field in the other two?